vllm/v1/engine/output_processor.py

```diff
@@ -74,6 +85,94 @@ def from_new_request(
             log_stats=log_stats,
         )
 
+    def make_request_output(
+        self,
+        new_token_ids: List[int],
+        finish_reason: Optional[FinishReason],
+        stop_reason: Union[int, str, None],
+    ) -> Optional[RequestOutput]:
+
+        finished = finish_reason is not None
+        output_kind = self.output_kind
+
+        # In follow up, we will switch to invariant where EngineCore
+        # does not stream partial prefills.
+        if not finished and (self.is_prefilling
+                             or output_kind == RequestOutputKind.FINAL_ONLY):
+            # Only the final output is required in FINAL_ONLY mode.
+            return None
+
+        # Aggregate child completions when not streaming
+        aggregate_children = output_kind == RequestOutputKind.FINAL_ONLY
+        request_output: Optional[RequestOutput] = None
+
+        if aggregate_children and self.output_aggregator is not None:
+            request_output = self.output_aggregator
+            self.output_aggregator = None
+        else:
+            request_output = self._new_request_output(
+                self.parent_req.request_id, finished)
+
+        completion_output = self._new_completion_output(
+            new_token_ids, finish_reason, stop_reason)
+        request_output.outputs.append(completion_output)
+
+        if (aggregate_children
+                and len(request_output.outputs) != self.parent_req.n):
+            self.output_aggregator = request_output
+            return None
+
+        return request_output
+
+    def _new_request_output(
+        self,
+        request_id: str,
+        finished: bool,
+    ) -> RequestOutput:
+
+        delta = self.output_kind == RequestOutputKind.DELTA
+
+        if delta:
```

Comment:
small simplification?
```suggestion
        if self.output_kind == RequestOutputKind.DELTA:
```